detector/calorimeter/HCalTileBarrel_o1_v01_geo.cpp

```diff
@@ -223,12 +223,39 @@ static dd4hep::Ref_t createHCal(dd4hep::Detector& lcdd, xml_det_t xmlDet, dd4hep
   }
   
   // Place envelope (or barrel) volume
-  Volume motherVol = lcdd.pickMotherVolume(hCal);
+  Volume motherVol = lcdd.pickMotherVolume(caloDetElem);
   motherVol.setVisAttributes(lcdd.invisible());
-  PlacedVolume placedHCal = motherVol.placeVolume(envelopeVolume);
-  placedHCal.addPhysVolID("system", hCal.id());
-  hCal.setPlacement(placedHCal);
-  return hCal;
+  PlacedVolume envelopePhysVol = motherVol.placeVolume(envelopeVolume);
+  envelopePhysVol.addPhysVolID("system", xmlDet.id());
```

Comment:
just to be sure, since you modified this: do xmlDet.id() and DetElement caloDetElem(xmlDet.nameStr(), xmlDet.id()).id() return the same thing?

Reply:
they do return the same thing, but just to be on the safe side I replaced xmlDet.id() by caloDetElem.id()